core/lens/lens_option.mli

```diff
@@ -0,0 +1,28 @@
+type 'a t = 'a option
```

Comment:
As per the above comment, I think this module is unnecessary.

Reply:
See above comment regarding `List` module.